presto-main/src/test/java/com/facebook/presto/operator/aggregation/differentialentropy/TestWeightedReservoirAggregation.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.differentialentropy;
+
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.metadata.MetadataManager;
+import com.facebook.presto.operator.aggregation.InternalAggregationFunction;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Locale;
+
+import static com.facebook.presto.block.BlockAssertions.createDoublesBlock;
+import static com.facebook.presto.block.BlockAssertions.createLongsBlock;
+import static com.facebook.presto.block.BlockAssertions.createRLEBlock;
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestWeightedReservoirAggregation
+        extends TestReservoirAggregation
+{
+    private InternalAggregationFunction function;
+
+    @BeforeClass
+    public void setUp()
+    {
+        FunctionManager functionManager = MetadataManager.createTestMetadataManager().getFunctionManager();
+        function = functionManager.getAggregateFunctionImplementation(
+                functionManager.lookupFunction(
+                        "differential_entropy",
+                        fromTypes(BIGINT, DOUBLE, DOUBLE)));
+    }
+
+    @Test
+    public void testBadNumberOfSamples()
```

Comment:
- use `expectedExceptions` and `expectedExceptionsMessageRegExp`
- remove unnecessary Long.valueOf and Double.valueOf
- rename to testInvalidMaxSamples (and testInvalidWeight)

```
    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Sample size -200 must be positive")
    public void testInvalidMaxSamples()
    {
        assertAggregation(
                getFunction(),
                0.0,
                createLongsBlock(-200),
                createDoublesBlock(0.1),
                createDoublesBlock(0.2));
    }

    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Weight -0.2 must be non-negative")
    public void testInvalidWeight()
    {
        assertAggregation(
                getFunction(),
                0.0,
                createLongsBlock(200),
                createDoublesBlock(0.1),
                createDoublesBlock(-0.2));
    }
```

Reply:
@mbasmanova I tried using the form you used above, but it's failing for me, since `assertAggregation(` is catching the exception and failing (it doesn't propagate to the decorator). Could you please advise? Thanks!

Reply:
@atavory The following works for me:

```
    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "maxSamples -200 must be positive")
    public void testBadNumberOfSamples()
    {
        assertAggregation(
                this.function,
                0.0,
                createLongsBlock(-200),
                createDoublesBlock(0.1),
                createDoublesBlock(0.2));
    }
```

Could you share the version that doesn't work along with the output of the test?

Reply:
Updated the PR's code. Here's the output
![image](https://user-images.githubusercontent.com/7824605/64379516-c184df00-d037-11e9-84dc-ef1ee85b8e37.png)

Reply:
@atavory The function being tested requires 6 parameters, but only 3 are being provided. Here is a fixed version:

```
    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "bucketCount -200 must be non-negative")
    public void testBadNumberOfSamples()
    {
        assertAggregation(
                getFunction(),
                0.0,
                createLongsBlock(-200),
                createDoublesBlock(0.1),
                createDoublesBlock(0.2),
                createStringsBlock(method),
                createDoublesBlock(0.0),
                createDoublesBlock(0.2));
    }
```